base_view_inheritance_extension/tests/test_base_view_inheritance_extension.py

```diff
@@ -0,0 +1,180 @@
+        source = etree.fromstring(
+            """
+            <form>
+                <field
+                    name="invoice_line_ids"
+                    context="{
+                        'default_type': context.get('default_type'),
+                        'journal_id': journal_id,
+                        'default_partner_id': commercial_partner_id,
+                        'default_currency_id': (
+                            currency_id != company_currency_id and currency_id or False
+                        ),
+                        'default_name': 'The company name',
+                    }"
+                />
+            </form>
+            """
+        )
+        specs = etree.fromstring(
+            """
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context" operation="update">
+                    {
+                        "default_product_id": product_id,
+                        "default_cost_center_id": (
+                            context.get("handle_mrp_cost") and cost_center_id or False
+                        ),
+                    }
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        expected_items = [
+            "'default_type': context.get('default_type')",
+            "'journal_id': journal_id",
+            "'default_partner_id': commercial_partner_id",
+            (
+                "'default_currency_id': "
+                "currency_id != company_currency_id and currency_id or False"
+            ),
+            "'default_name': 'The company name'",
+            "'default_product_id': product_id",
+            (
+                "'default_cost_center_id': "
+                "context.get('handle_mrp_cost') and cost_center_id or False"
+            ),
+        ]
+        self.assertEqual(
+            res.xpath('//field[@name="invoice_line_ids"]')[0].attrib["context"],
+            "{%s}" % ", ".join(expected_items),  # noqa: UP031
+        )
+
+    def test_update_empty_source_dict(self):
+        """Test that we can add new keys by creating the dict if it's missing"""
+        source = etree.fromstring(
+            """
+            <form>
+                <field name="ref" />
+            </form>
+            """
+        )
+        specs = etree.fromstring(
+            """
+            <field name="ref" position="attributes">
+                <attribute name="context" operation="update">
+                    {
+                        "default_company": 1,
+                    }
+                </attribute>
+            </field>
+            """
+        )
+        res = self.env["ir.ui.view"].apply_inheritance_specs(source, specs)
+        self.assertEqual(
+            res.xpath('//field[@name="ref"]')[0].attrib["context"],
+            "{'default_company': 1}",
+        )
+
+    def test_update_operation_not_a_dict(self):
```

Comment:
As you mentioned above. Use **`domain`** or something else to replace **`attrs`**

Reply:
Let's take another example. This example can also be done using odoo native if attribute is missing.

Reply:
any update?

Reply:
I use `domain` to replace `attrs`

Reply:
> Let's take another example. This example can also be done using odoo native if attribute is missing.

Hi @LucasTran380381 
Nils is referring to this comment of mine :wink: